doc/OnlineDocs/contributed_packages/mindtpy.rst

```diff
@@ -71,67 +107,179 @@ MindtPy also supports setting options for mip solver and nlp solver.
+
+.. | constraint | c_geq | status at x1 | tmp_dual (violation) |
+.. |------------|-------|--------------|----------------------|
+.. | g(x) <= b  | -1    | g(x1) <= b   | 0                    |
+.. | g(x) <= b  | -1    | g(x1) > b    | g(x1) - b            |
+.. | g(x) >= b  | +1    | g(x1) >= b   | 0                    |
+.. | g(x) >= b  | +1    | g(x1) < b    | b - g(x1)            |
+
+Augmented Penalty
+-----------------
+
+Augmented Penalty refers to the introduction of (non-negative) slack variables on the right hand sides of the just described inequality constraints and the modification of the objective function when assumptions concerning convexity do not hold. (From DICOPT)
+
+
 Global Outer Approximation
----------------------------------------------
+--------------------------
+
+Apart of the decomposition methods for convex MINLP problems [`Kronqvist et al., 2019`_], MindtPy provides an implementation of Global Outer Approximation (GOA) as described in [`Kesavan et al., 2004`_], to provide optimality guaranteed for nonconvex MINLP problems. Here, the validity of the Mixed-integer Linear Programming relaxation of the original problem is guaranteed via the usage of Generalized McCormick envelopes, computed using the package `MC++`_. The NLP subproblems in this case need to be solved to global optimality, which can be achieved through global NLP solvers such as `BARON`_ or `SCIP`_. 
+
+.. _BARON: https://minlp.com/baron-solver
+.. _SCIP: https://www.scipopt.org/
+
 
-Apart of the decomposition methods for convex MINLP problems [`Kronqvist et al., 2019`_], MindtPy provides an implementation of Global Outer Approximation (GOA) as described in [`Kesavan et al., 2004`_], to provide optimality guaranteed for nonconvex MINLP problems. Here, the validity of the Mixed-integer Linear Programming relaxation of the original problem is guaranteed via the usage of Generalized McCormick envelopes, computed using the package `MC++`_. The NLP subproblems in this case need to be solved to global optimality, which can be achieved through global NLP solvers such as BARON or SCIP. No-good cuts are added to each iteration, guaranteeing the finite convergence of the algorithm. Notice that this method is more computationally expensive than the other strategies implemented for convex MINLP like OA and ECP, which in turn can be used as heuristics for nonconvex MINLP problems.
+Convergence
+^^^^^^^^^^^
+
+MindtPy provides two ways to guarantee the finite convergence of the algorithm. 
+
+- **No-good cuts**. No-good cuts(integer cuts) are added to the MILP master problem in each iteration. 
+- **Tabu list**. Tabu list is supported only if the ``mip_solver`` is ``cplex_persistent`` (``gurobi_persistent`` pending). In each iteration, the explored integer combinations will added to the tabu_list. When solving the next MILP problem, the MIP solver will reject the previously explored solutions in the branch and bound process through IncumbentCallback.
```

Comment:
- **Tabu list**. Tabu list is only supported if the ``mip_solver`` is ``cplex_persistent`` (``gurobi_persistent`` pending). In each iteration, the explored integer combinations will be added to the `tabu_list`. When solving the next MILP problem, the MIP solver will reject the previously explored solutions in the branch and bound process through IncumbentCallback.